packages/@azure/servicebus/data-plane/test/streamingReceiverSessions.spec.ts

```diff
@@ -785,10 +785,10 @@ describe("Streaming Receiver - Settle an already Settled message throws error(wi
       return Promise.resolve();
     }, unExpectedErrorHandler);
 
-    await delay(5000);
+    const msgsCheck = await DelayStreaming(() => receivedMsgs.length === 1);
+    should.equal(msgsCheck, true, "Could not receive the messages in expected time.");
     should.equal(unexpectedError, undefined, unexpectedError && unexpectedError.message);
 
-    should.equal(receivedMsgs.length, 1);
```

Comment:
Maybe we shouldnt remove this check.
What if we got more than 1? then the test should fail isnt it?

Reply:
Same for other cases where we are removing the check

Reply:
The only way we can get more than 1 is we receive a message after `receivedMsgs.length === 1` is verified.
Then, we should also give more scope to receive messages - instead of returning `true` right after the  `receivedMsgs.length === 1` check, we can add `await delay(retry)`[in the DelayStreaming module] and then return `true`.
Does this make sense ?

Reply:
Ah, thats ok, we can keep it the way it is

Reply:
@ramya-rao-a You want me keep this `should.equal(receivedMsgs.length, 1);` ?

Reply:
Github is acting weird... I am adding a reply to one comment thread.. it is appearing in another :)

When waiting for the 1 sec inside the loop, it can happen that the receiver gets the second message. So, we should keep the `should.equal(receivedMsgs.length, 1)` check

Reply:
We are not waiting for 1 sec inside the loop.
```typescript
export async function checkWithTimeout(
  predicate: () => boolean,
  delayBetweenRetriesInMilliseconds: number = 1000,
  maxWaitTimeInMilliseconds: number = 10000
): Promise<boolean> {
  const maxTime = Date.now() + maxWaitTimeInMilliseconds;
  while (Date.now() < maxTime) {
    if (predicate()) return true;
    await delay(delayBetweenRetriesInMilliseconds);
  }
  return false;
}
```
- We return `true` as soon as `receivedMsgs.length === 1` happens.
- The only way we can get more than 1 message is if we receive a message after `receivedMsgs.length === 1` is verified.

----------------
I suggest we add a `await delay(delayBetweenRetriesInMilliseconds);`
  just before we `return true` and then keep the check `should.equal(receivedMsgs.length, 1);` in the tests.

Updated checkWithTimeout looks like
```typescript
export async function checkWithTimeout(
  predicate: () => boolean,
  delayBetweenRetriesInMilliseconds: number = 1000,
  maxWaitTimeInMilliseconds: number = 10000
): Promise<boolean> {
  const maxTime = Date.now() + maxWaitTimeInMilliseconds;
  while (Date.now() < maxTime) {
    if (predicate()){
        await delay(delayBetweenRetriesInMilliseconds);
        return true;
    } 
    await delay(delayBetweenRetriesInMilliseconds);
  }
  return false;
}
```
My only concern with this is, it basically increases the the time taken by each test by 1 second.

Reply:
`checkWithTimeout()` should definitely return immediately when the predicate is true.